keps/693-multikueue/README.md

```diff
@@ -0,0 +1,225 @@
+
+![Architecture](arch.png "Architecture")
+
+For each workload coming to a ClusterQueue (with the MultiKueue AdmissonCheck enabled)
+in the management cluster, and getting past the preadmission phase in the 
+two-phase admission process (meaning that the global quota is ok), 
+MultiKueue controller will create it in the defined worker clusters and wait 
+until some Kueue running there admits the workload. 
+Then it will remove the workloads from the remaining clusters and allow the
+single instance of the job to proceed. The workload will be also admitted in 
+the management cluster.
+
+There will be no job controllers running in the management clusters or they will be
+disabled for the workloads coming to MultiKueue-enabled cluster queues via annotation
+or some other, yet to be decided, mechanism. By disabling we mean that the controller
+will do no action on the selected objects, no pods (or other objects created and 
+allowing other controllets to update their status as they see fit.
+
+There will be just CRD/job definitions deployed. MultiKueue controller will copy the status
+of the job from the worker clusters, so that it will appear that the job
+is running inside the management clusters. However, as there is no job controller,
+no pods will be created in the management cluster. Noone will also overwrite
+the status that will be copied by the MultiKueue controller. 
+
+If the job, for whatever reason, is suspended or deleted in the management cluster,
+it will be deleted from the worker cluster. Deletion/suspension of the job 
+only in worker cluster will trigger the global job requeuing.
+Once the job finishes in the worker cluster, the job will also 
+finish in the management cluster.
+
+### User Stories (Optional)
+
+#### Story 1
+As a Kueue user I have clusters on different cloud providers and on-prem. 
+I would like to run computation-heavy jobs across all of them, wherever 
+I have free resources.
+ 
+#### Story 2
+As a Kueue user I have clusters in multiple regions of the same cloud 
+provider. I would like to run workloads that require the newest GPUs,
+whose on-demand availability is very volatile. The GPUs are available
+at random times at random regions. I want to use ProvisioningRequest
+to try to catch them.
+
+### Risks and Mitigations
+* Disabling the Job controller for all (or selected objects) may be problematic
+on environments where access to the master configuration is limited (like GKE).
+We are in talks with SIG-Apps to establish an acceptable way of using a non
+default controller (or none at all)
+
+## Design Details
+MultiKueue will be enabled on a cluster queue using the admission check fields.
+Just like ProvisioningRequest, MultiKueue will have its own configuration, 
+MultiKueueConfig with the following definition. To allow reusig the same clusters
+across many Kueues, additional object, MultiKueueWorkerCluster, is added.
+
+```go
+type MultiKueueConfig struct {
```

Comment:
@mwielgus @trasc I would like to see some alignment of the API between the KEP and the WIP implementation, currently the types are disjoint. Possibly they correspond to the same concepts, but it's not obvious. 

Implementation types for reference: https://github.com/kubernetes-sigs/kueue/blob/e8152a548404ad18a2cd1f78022b271ddf87b627/apis/kueue/v1beta1/delegateconfig_types.go#L24-L70.

Reply:
I'd say that MultiKueueConfig makes more sense, just like we named the other one ProvisioningRequestConfig.

Also, in Marcin's proposal, the Clusters themselves are a separate object, which allows for reutilization in different ClusterQueues, so that it's easier to point to a subset of the clusters.

@trasc please update your PR.

Reply:
From a configuration point of view , indeed the list of references is a plus when reusing, but from a runtime point view is not ideal, when a worker gets reconciled we can get to the root config with 2 API reads , with the reference reads an additional API read is needed for each configured cluster. 

I hote I'll manage to do the name sync by EOW and maybe we can have another look.

Reply:
I've done the rename https://github.com/kubernetes-sigs/kueue/blob/ba0d211d358f00417c8d80834a3cadd0e7c8c5a6/apis/kueue/v1beta1/multikueueconfig_types.go.

Going forward, I think it will be a lot easier to  just have multiple "MultiKueue" admission check added to a cluster queue each of them containing a single destination cluster.

Reply:
The admission check for a worker that is not selected for execution can be set to ready once the cleanup is done in it with a message indicating that the execution will take place in another worker.